src/java.base/share/classes/sun/security/util/Pem.java

```diff
@@ -25,15 +25,93 @@
 
 package sun.security.util;
 
-import java.io.IOException;
+import sun.security.x509.AlgorithmId;
+
+import java.io.*;
 import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+import java.util.Arrays;
 import java.util.Base64;
+import java.util.HexFormat;
+import java.util.Objects;
 
 /**
  * A utility class for PEM format encoding.
  */
 public class Pem {
 
+    /**
+     * Public Key PEM header & footer
+     */
+    public static final byte[] PUBHEADER = "-----BEGIN PUBLIC KEY-----"
+        .getBytes(StandardCharsets.UTF_8);
+    public static final byte[] PUBFOOTER = "-----END PUBLIC KEY-----"
+        .getBytes(StandardCharsets.UTF_8);
+
+    /**
+     * Private Key PEM header & footer
+     */
+    public static final byte[] PKCS8HEADER = "-----BEGIN PRIVATE KEY-----"
+        .getBytes(StandardCharsets.UTF_8);
+    public static final byte[] PKCS8FOOTER = "-----END PRIVATE KEY-----"
+        .getBytes(StandardCharsets.UTF_8);
+
+    /**
+     * Encrypted Private Key PEM header & footer
+     */
+    public static final byte[] PKCS8ENCHEADER = "-----BEGIN ENCRYPTED PRIVATE KEY-----"
+        .getBytes(StandardCharsets.UTF_8);
+    public static final byte[] PKCS8ENCFOOTER = "-----END ENCRYPTED PRIVATE KEY-----"
+        .getBytes(StandardCharsets.UTF_8);
+
+    /**
+     * Certificate PEM header & footer
+     */
+    public static final byte[] CERTHEADER = "-----BEGIN CERTIFICATE-----"
+        .getBytes(StandardCharsets.UTF_8);
+    public static final byte[] CERTFOOTER = "-----END CERTIFICATE-----"
+        .getBytes(StandardCharsets.UTF_8);
+
+    /**
+     * CRL PEM header & footer
+     */
+    public static final byte[] CRLHEADER = "-----BEGIN CRL-----"
+        .getBytes(StandardCharsets.UTF_8);
+    public static final byte[] CRLFOOTER = "-----END CRL-----"
+        .getBytes(StandardCharsets.UTF_8);
+
+    /**
+     * PKCS#1/slleay/OpenSSL RSA PEM header & footer
+     */
+    public static final byte[] PKCS1HEADER = "-----BEGIN RSA PRIVATE KEY-----"
+        .getBytes(StandardCharsets.UTF_8);
+    public static final byte[] PKCS1FOOTER = "-----END RSA PRIVATE KEY-----"
+        .getBytes(StandardCharsets.UTF_8);
+
+    public static final byte[] LINESEPARATOR = "\r\n"
+        .getBytes(StandardCharsets.UTF_8);
```

Comment:
These arrays should probably be marked as [`@Stable`]:
```suggestion
    /**
     * Public Key PEM header & footer
     */
    public static final @Stable byte[] PUBHEADER = "-----BEGIN PUBLIC KEY-----"
        .getBytes(StandardCharsets.UTF_8);
    public static final @Stable byte[] PUBFOOTER = "-----END PUBLIC KEY-----"
        .getBytes(StandardCharsets.UTF_8);

    /**
     * Private Key PEM header & footer
     */
    public static final @Stable byte[] PKCS8HEADER = "-----BEGIN PRIVATE KEY-----"
        .getBytes(StandardCharsets.UTF_8);
    public static final @Stable byte[] PKCS8FOOTER = "-----END PRIVATE KEY-----"
        .getBytes(StandardCharsets.UTF_8);

    /**
     * Encrypted Private Key PEM header & footer
     */
    public static final @Stable byte[] PKCS8ENCHEADER = "-----BEGIN ENCRYPTED PRIVATE KEY-----"
        .getBytes(StandardCharsets.UTF_8);
    public static final @Stable byte[] PKCS8ENCFOOTER = "-----END ENCRYPTED PRIVATE KEY-----"
        .getBytes(StandardCharsets.UTF_8);

    /**
     * Certificate PEM header & footer
     */
    public static final @Stable byte[] CERTHEADER = "-----BEGIN CERTIFICATE-----"
        .getBytes(StandardCharsets.UTF_8);
    public static final @Stable byte[] CERTFOOTER = "-----END CERTIFICATE-----"
        .getBytes(StandardCharsets.UTF_8);

    /**
     * CRL PEM header & footer
     */
    public static final @Stable byte[] CRLHEADER = "-----BEGIN CRL-----"
        .getBytes(StandardCharsets.UTF_8);
    public static final @Stable byte[] CRLFOOTER = "-----END CRL-----"
        .getBytes(StandardCharsets.UTF_8);

    /**
     * PKCS#1/slleay/OpenSSL RSA PEM header & footer
     */
    public static final @Stable byte[] PKCS1HEADER = "-----BEGIN RSA PRIVATE KEY-----"
        .getBytes(StandardCharsets.UTF_8);
    public static final @Stable byte[] PKCS1FOOTER = "-----END RSA PRIVATE KEY-----"
        .getBytes(StandardCharsets.UTF_8);

    public static final @Stable byte[] LINESEPARATOR = "\r\n"
        .getBytes(StandardCharsets.UTF_8);
```

[`@Stable`]: https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/jdk/internal/vm/annotation/Stable.java

Reply:
I have not seen `@Stable` before. I will have to evaluate this suggestion further

Reply:
Much of these got remove in an update